### PR TITLE
Reader Following Manage: show email settings immediately on follow

### LINKED ISSUE #14069: Reader: Settings for followed Site Rec only appears after refresh
Reported by @designsimply.

Steps to repro:
1. Go to https://wpcalypso.wordpress.com/following/manage.
2. Follow a site from Recommended Sites.
3. This site gets added to your Following list, but the Settings button only shows up after refreshing.

screen-shot-2017-05-15-at-mon-may-15-3-24-41-pm

### PR DESCRIPTION
**Pull request by bluefuton**
We currently don't show the 'Settings' link for a new follow until after a page reload.

Fixes #14069.

Todo:
- [x] correct following status so link is shown
- [ ] show correct initial settings in settings dialog

### CODE REVIEW

File: client/blocks/reader-subscription-list-item/index.jsx [hunk 1]
```diff
@@ -53,7 +53,6 @@ function ReaderSubscriptionListItem( {
 	const streamUrl = getStreamUrl( feedId, siteId );
 	const feedUrl = url || getFeedUrl( { feed, site } );
 	const siteUrl = getSiteUrl( { feed, site } );
-	const isFollowing = ( site && site.is_following ) || ( feed && feed.is_following );
```
Comment (blowery) [inline client/blocks/reader-subscription-list-item/index.jsx]: goooooood catch. yikes.

maybe we should poke around and see if we're using site or feed.is_following anywhere else...

Comment (blowery) [inline client/blocks/reader-subscription-list-item/index.jsx]: or maybe we should start modifying loaded site and feed objects on follow / unfollow...

File: client/reader/following-manage/connected-subscription-list-item.jsx [hunk 2]
```diff
@@ -39,7 +42,16 @@ class ConnectedSubscriptionListItem extends React.Component {
 	}
 
 	render() {
-		const { feed, site, translate, url, feedId, siteId, showEmailSettings } = this.props;
+		const {
+			feed,
```
Comment (blowery) [inline client/reader/following-manage/connected-subscription-list-item.jsx]: so much more readable. yay prettier.

File: client/reader/following-manage/connected-subscription-list-item.jsx [hunk 3]
```diff
@@ -51,9 +63,12 @@ class ConnectedSubscriptionListItem extends React.Component {
...
 				showEmailSettings={ showEmailSettings && ! isEmailBlocked }
+				isFollowing={ isFollowing }
 			/>
 		);
 	}
 }
 
-export default localize( connectSite( ConnectedSubscriptionListItem ) );
+export default connect( ( state, ownProps ) => ( {
+	isFollowing: isFollowingSelector( state, { feedId: ownProps.feedId, blogId: ownProps.siteId } ),
```
Comment (blowery) [inline client/reader/following-manage/connected-subscription-list-item.jsx]: this should probably use the url prop we have. i'm fairly sure that's a feed url?

Comment (bluefuton) [inline client/reader/following-manage/connected-subscription-list-item.jsx]: Ah, interesting! I thought using the IDs might be more reliable. Why's it best to use URL in preference?

**Comment by samouri** [comment]
I think this was based off the blacklisting descriptions branch instead of master

**Comment by bluefuton** [comment]
> I think this was based off the blacklisting descriptions branch instead of master

Something went very wrong with that branch - managed to force push 0 commits in the end, which closed the PR. Yay. Continuing in https://github.com/Automattic/wp-calypso/pull/14188.

**Pull request closed without merging**